sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/coverage/internal/DefaultCoverage.java

```diff
@@ -128,6 +147,7 @@ public int coveredConditions() {
   public void doSave() {
     Preconditions.checkNotNull(inputFile, "Call onFile() first");
     Preconditions.checkNotNull(type, "Call ofType() first");
+    validateLines();
```

Comment:
I prefer to enforce some ordering in the fluent API like it was done in other similar API. Just fail if lineHits() is called while onFile() was not already called.